File: app/src/androidTest/java/de/westnordost/streetcomplete/quests/AddBuildingLevelsTest.java
package de.westnordost.streetcomplete.quests;

import de.westnordost.streetcomplete.data.osm.OsmElementQuestType;
import de.westnordost.streetcomplete.data.osm.changes.StringMapEntryAdd;
import de.westnordost.streetcomplete.quests.building_levels.AddBuildingLevels;
import de.westnordost.streetcomplete.quests.building_levels.AddBuildingLevelsForm;

public class AddBuildingLevelsTest extends AOsmElementQuestTypeTest
{
	@Override public void setUp() throws Exception
	{
		super.setUp();
		tags.put("building","residential");
	}

	public void testBuildingLevelsOnly()
	{
		bundle.putInt(AddBuildingLevelsForm.BUILDING_LEVELS, 5);
		verify(
				new StringMapEntryAdd("building:levels","5"));
	}

	public void testBuildingLevelsAndZeroRoofLevels()
	{
		bundle.putInt(AddBuildingLevelsForm.BUILDING_LEVELS, 5);
		bundle.putInt(AddBuildingLevelsForm.ROOF_LEVELS, 0);
		verify(
				new StringMapEntryAdd("building:levels","5"),
				new StringMapEntryAdd("roof:levels","0"));
	}

	public void testBuildingLevelsAndRoofLevels()
	{
		bundle.putInt(AddBuildingLevelsForm.BUILDING_LEVELS, 5);
		bundle.putInt(AddBuildingLevelsForm.ROOF_LEVELS, 3);
		verify(
				new StringMapEntryAdd("building:levels","5"),
				new StringMapEntryAdd("roof:levels","3"));
	}

	@Override protected OsmElementQuestType createQuestType()
	{
		return new AddBuildingLevels(null);
	}
}

File: app/src/androidTest/java/de/westnordost/streetcomplete/quests/AddBuildingLevelsTest.kt
package de.westnordost.streetcomplete.quests

import de.westnordost.streetcomplete.data.osm.changes.StringMapEntryAdd
import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao
import de.westnordost.streetcomplete.quests.building_levels.AddBuildingLevels
import de.westnordost.streetcomplete.quests.building_levels.AddBuildingLevelsForm
import org.junit.Before
import org.junit.Test

import org.mockito.Mockito.mock

class AddBuildingLevelsTest : AOsmElementQuestTypeTest() {

    override val questType = AddBuildingLevels(mock(OverpassMapDataDao::class.java))

    @Test fun buildingLevelsOnly() {
        bundle.putInt(AddBuildingLevelsForm.BUILDING_LEVELS, 5)
        verify(
            StringMapEntryAdd("building:levels", "5")
        )
    }

    @Test fun buildingLevelsAndZeroRoofLevels() {
        bundle.putInt(AddBuildingLevelsForm.BUILDING_LEVELS, 5)
        bundle.putInt(AddBuildingLevelsForm.ROOF_LEVELS, 0)
        verify(
            StringMapEntryAdd("building:levels", "5"),
            StringMapEntryAdd("roof:levels", "0")
        )
    }

    @Test fun buildingLevelsAndRoofLevels() {
        bundle.putInt(AddBuildingLevelsForm.BUILDING_LEVELS, 5)
        bundle.putInt(AddBuildingLevelsForm.ROOF_LEVELS, 3)
        verify(
            StringMapEntryAdd("building:levels", "5"),
            StringMapEntryAdd("roof:levels", "3")
        )
    }
}
